pkg/controller/job/job_controller_test.go

```diff
@@ -1827,40 +1827,40 @@ func hasTrueCondition(job *batch.Job) *batch.JobConditionType {
 }
 
 func TestSyncPastDeadlineJobFinished(t *testing.T) {
```

Comment:
This test doesn't proof your fix is working as expected, with or without the fix the test passes. 
I think we'd need 2 tests:
1. the current one
2. the one where you have `job.Status.StartTime` set when creating a job, this mimics the behavior you're seeing.

Reply:
You transform this test into a test table, where a boolean controls whether StartTime is set

Reply:
Ack. Let me push an updated test for this by EOD.

Reply:
Updated the changes for tests. PTAL. cc @alculquicondor  @soltysh